packages/presentation/lib/ui/auth_page/bloc/auth_bloc.dart

```diff
@@ -0,0 +1,106 @@
+import 'package:domain/model/firebase_user_email.dart';
+import 'package:domain/use_case/auth_use_case.dart';
+import 'package:domain/use_case/login_facebook_use_case.dart';
+import 'package:domain/use_case/login_google_use_case.dart';
+import 'package:flutter/material.dart';
+import 'package:presentation/base/bloc.dart';
+import 'package:presentation/navigation/base_arguments.dart';
+import 'package:presentation/services/firebase_analytics.dart';
+import 'package:presentation/ui/auth_page/bloc/auth_tile.dart';
+import 'package:presentation/ui/auth_page/bloc/validator/validator.dart';
+import 'package:presentation/ui/profile_page/profile_widget.dart';
+
+abstract class AuthBloc extends Bloc<BaseArguments, AuthTile> {
+  factory AuthBloc(
+    LoginEmailAndPassUseCase authUseCase,
+    Analytics analytics,
+    LoginGoogleUseCase loginGoogleUseCase,
+    LoginFacebookUseCase loginFacebookUseCase,
+  ) =>
+      AuthBlocImpl(
+        authUseCase: authUseCase,
+        analytics: analytics,
+        loginGoogleUseCase: loginGoogleUseCase,
+        loginFacebookUseCase: loginFacebookUseCase,
+      );
+
+  TextEditingController get textLoginController;
+
+  TextEditingController get textPasswordController;
+
+  Future<void> auth();
+
+  Future<void> authFacebook();
+
+  Future<void> authGoogle();
+}
+
+class AuthBlocImpl extends BlocImpl<BaseArguments, AuthTile>
+    implements AuthBloc {
+  var _tile = AuthTile.init();
+  final _loginController = TextEditingController();
+  final _passwordController = TextEditingController();
+  final LoginGoogleUseCase loginGoogleUseCase;
+  final LoginFacebookUseCase loginFacebookUseCase;
+  final LoginEmailAndPassUseCase authUseCase;
+  final Analytics analytics;
+
+  @override
+  TextEditingController get textLoginController => _loginController;
+
+  @override
+  TextEditingController get textPasswordController => _passwordController;
+
+  AuthBlocImpl({
+    required this.authUseCase,
+    required this.analytics,
+    required this.loginGoogleUseCase,
+    required this.loginFacebookUseCase,
+  });
+
+  @override
+  Future<void> authFacebook() async {
+    analytics.analyticsFacebookClick();
+    await _tryLogin(await loginFacebookUseCase());
+  }
+
+  @override
+  Future<void> authGoogle() async {
+    analytics.analyticsGoogleClick();
+    _tryLogin(await loginGoogleUseCase());
+  }
+
+  @override
+  Future<void> auth() async {
+    final login = _loginController.text;
+    final password = _passwordController.text;
```

Comment:
use onChange of TextField vs controller in this case

Reply:
why? in this case, the controller text suits us much more, since we don’t care about changing the number of values ​​at the moment, onChanged mainly used for searching

Reply:
agree, but at the same time, bloc should keep our current state. so, when user will click the btn, we would already have current state in bloc, but not getting it by controller